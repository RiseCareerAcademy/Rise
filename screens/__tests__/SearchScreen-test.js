import 'react-native';
import React from 'react';
import Search from '../SearchScreen';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const props = {
      navigation: {
        getParam: () => {
            return 'Tracy';
        },
          navigate: () => {},
      }
  };
  const tree = renderer.create(<Search {...props}>Snapshot test!</Search>).toJSON();

  expect(tree).toMatchSnapshot();

});

it('returns scores and matches on Tracy', () => {
    const props = {
      navigation: {
        getParam: () => {
            return 'Tracy';
        },
        navigate: () => { }
      },
    };
    const instance = renderer.create(<Search {...props} />).getInstance();
    expect(instance.search( [
        ["Daniel Ng"],
        ["Software Developer"],
        ["JQuery", "Python", "UX"], //should be number 4
        ["Tracy Lewis"],
        [""],
        ["UX", "Prototyping"], //should be number 3
        ["Kevin Mui"],
        ["Product Manager"],
        [""], //should be number 2
        ["Lewis Tracy"],
        ["Product Manager"],
        ["UX"], //should be number 1
        ["Tone Yu"],
        ["Nurse"],
        ["Birthing", "Yelling", "Running"] //should not be in the results
      ])).toEqual({"matches": [["Tracy Lewis"], ["Lewis Tracy"]], "scores": [2, 2]},['Tracey Lewis','Lewis Tracy']);
  });
  it('returns empty scores and matches on empty input', () => {
    const props = {
      navigation: {
        getParam: () => {
            return '';
        },
        navigate: () => { }
      },
    };
    const instance = renderer.create(<Search {...props} />).getInstance();
    expect(instance.search( [
        ["Daniel Ng"],
        ["Software Developer"],
        ["JQuery", "Python", "UX"], //should be number 4
        ["Tracy Lewis"],
        [""],
        ["UX", "Prototyping"], //should be number 3
        ["Kevin Mui"],
        ["Product Manager"],
        [""], //should be number 2
        ["Lewis Tracy"],
        ["Product Manager"],
        ["UX"], //should be number 1
        ["Tone Yu"],
        ["Nurse"],
        ["Birthing", "Yelling", "Running"] //should not be in the results
      ])).toEqual({"matches": [], "scores": []});
  });
  it('returns empty scores and matches on input Tracy and empty database', () => {
    const props = {
      navigation: {
        getParam: () => {
            return 'Tracy';
        },
        navigate: () => { }
      },
    };
    const instance = renderer.create(<Search {...props} />).getInstance();
    expect(instance.search([])).toEqual({"matches": [], "scores": []});
  });