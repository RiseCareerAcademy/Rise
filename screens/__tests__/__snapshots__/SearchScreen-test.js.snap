// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <Styled(Content)
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
  >
    <Search
      cancelButtonTitle="Cancel"
      icon={Object {}}
      lightTheme={false}
      loadingIcon={Object {}}
      noIcon={false}
      onChangeText={[Function]}
      placeholder="Search"
      placeholderTextColor="#86939e"
      platform="ios"
      round={false}
      showLoading={true}
      showLoadingIcon={false}
    />
    <CheckBox
      center={false}
      checked={true}
      checkedColor="green"
      checkedIcon="check-square-o"
      iconRight={false}
      onPress={[Function]}
      right={false}
      size={24}
      title="Matches"
      uncheckedColor="#bfbfbf"
      uncheckedIcon="square-o"
    />
    <CheckBox
      center={false}
      checked={true}
      checkedColor="green"
      checkedIcon="check-square-o"
      iconRight={false}
      onPress={[Function]}
      right={false}
      size={24}
      title="Mentors"
      uncheckedColor="#bfbfbf"
      uncheckedIcon="square-o"
    />
    <CheckBox
      center={false}
      checked={true}
      checkedColor="green"
      checkedIcon="check-square-o"
      iconRight={false}
      onPress={[Function]}
      right={false}
      size={24}
      title="Mentees"
      uncheckedColor="#bfbfbf"
      uncheckedIcon="square-o"
    />
    <Styled(Content)
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
    />
  </Styled(Content)>
</Component>
`;
